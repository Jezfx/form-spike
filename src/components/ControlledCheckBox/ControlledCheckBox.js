import React from "react";
import { Controller, useFormContext } from "react-hook-form";
import {
  Checkbox,
  FormControlLabel,
  FormControl,
  FormHelperText,
  FormLabel,
  FormGroup,
} from "@material-ui/core";

const ControlledCheckBox = ({ name, label, options = [] }) => {
  const { control, errors } = useFormContext();
  const error = errors[name];

  return (
    <FormControl component="fieldset" error={!!error}>
      <FormLabel component="legend">{label}</FormLabel>
      <FormGroup>
        {options?.map((props, index) => (
          <FormControlLabel
            key={`${name}[${index}]`}
            label={props.label}
            control={
              <Controller
                name={`${name}[${index}]`}
                control={control}
                render={({ onChange, name }) => (
                  <Checkbox
                    name={`${name}[${index}]`}
                    value={`${name}[${props.value}]`}
                    onChange={(e) => {
                      e.target.checked ? onChange(props.value) : onChange();
                    }}
                  />
                )}
              />
            }
          />
        ))}
      </FormGroup>
      <FormHelperText>{error?.message}</FormHelperText>
    </FormControl>
  );
};

export default ControlledCheckBox;
