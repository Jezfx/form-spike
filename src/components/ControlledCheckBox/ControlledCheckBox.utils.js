import { curry, flow, map, toLower, includes } from "lodash/fp";

export const isStringIncludedInArray = curry((currentValues, value) =>
  flow(map(toLower), includes(value))(currentValues)
);
