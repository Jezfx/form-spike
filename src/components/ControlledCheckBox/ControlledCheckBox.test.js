import React from "react";
import ControlledCheckBox from "./ControlledCheckBox";
import { renderWithReactHookForm } from "../../test-utils";

describe("ControlledCheckBox", () => {
  it("should match snapshot", () => {
    const { asFragment } = renderWithReactHookForm(<ControlledCheckBox />);

    expect(asFragment()).toMatchSnapshot();
  });
});
