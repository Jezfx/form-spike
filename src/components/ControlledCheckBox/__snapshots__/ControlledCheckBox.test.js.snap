// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledCheckBox should match snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root"
  >
    <p
      class="MuiFormHelperText-root"
    />
  </fieldset>
</DocumentFragment>
`;
